### DISCUSSION TITLE
easy install on win

### DISCUSSION THREAD

**Discussion by kalle07** [khoj-ai/khoj#795] [Q&A]
you have the intention to get your code running in a normal way?
no PgVector or Docker (both i dont want)
i mean all 10th local Ai software i know (like GPT4All, Jan, LM Studio ....) )all can do it ... all are open , you can copy and paste  ;)

**Comment by andreiramani** [comment]
if you want to run locally, you can read here [https://docs.khoj.dev/get-started/setup](https://docs.khoj.dev/get-started/setup)

**Comment by debanjum** [comment]
Yeah, as @andreiramani mentioned, you should be able to run Khoj on Window with Docker or Pip. See the updated [self-host docs](https://docs.khoj.dev/get-started/setup/?os=windows) for setup instructions.

@kalle07 the reason Khoj has a more complex local only install (using pgvector, a server-client architecture etc.) is because the application scales from a single-user, on-device, personal AI to a multi-user, multi-device, enterprise AI.

This allows folks to access Khoj across their Phone, Desktop, Obsidian or Emacs, create automations, try open & commerical LLMs etc.

But it doesn't currently make it easy for folks not used to a terminal to self-host Khoj. Hopefully the local setup gets easier as we continue to evolve Khoj. Ideas on how to do this are always welcome! 🙂

**Reply by STORMFIRE007** [reply]
one idea that i can think of to make the installation easier is perhaps somehow make the khoj installer automatically install it's prerequisites aswell, such as postgreSQL and pgvectors, personally i had a tough time building and installing pgvectors because of conflicts between different versions of Visual Studio, which caused the target machine to be detected as x86 ,but that might be too complex to set up

another way i could think of to simplify the install process is perhaps to have some simple defaults for each option (the LLM, speech ai etc), so if a user just presses enter and it proceeds with defaults 

or maybe to give user ability to choose an option that whatever model user wants to install, somewhat similar to how oobabooga textgen webui installer does, for example:

```
what model you wish to use:
1. LLaMa
2. bartowski/gemma-2-2b-it-GGUF
3. bartowski/Qwen2.5-3B-Instruct-GGUF
4. enter your own
```

so beginners can just enter 1,2 or 3, and press enter whereas the more advanced users can enter their own selection by entering 4

**Reply by andreiramani** [reply]
If you interested in pgvector, I try to make binary compiled pgvector for PostgreSQL 16 in Microsoft Windows 11 - https://github.com/andreiramani/pgvector_pgsql_windows

> one idea that i can think of to make the installation easier is perhaps somehow make the khoj installer automatically install it's prerequisites aswell, such as postgreSQL and pgvectors, personally i had a tough time building and installing pgvectors because of conflicts between different versions of Visual Studio, which caused the target machine to be detected as x86 ,but that might be too complex to set up
>

**Comment by kalle07** [comment]
i see...

i dont know secials
but i think LM-Studio and Antyhing-LLM has seperate this stuff in two programs (so you you can run the and listen to a serve rport)
https://lmstudio.ai/
https://anythingllm.com/

i know both are exe installers so very compfortable

**Comment by kalle07** [comment]
i see ... no easy way for install like

create venv (may you can download and install your viusalstudio stuff)
and 
pip install -r  ...

and thats it ?



is there a way to install only one part
and combine it with
https://anythingllm.com/desktop
